pennylane_lightning/core/_measurements_base.py

```diff
@@ -70,6 +70,17 @@ def dtype(self):
         """Returns the simulation data type."""
         return self._qubit_state.dtype
 
+    def _observable_is_sparse(self, obs):
+        """States if the required observable is sparse.
+
+        Args:
+            obs: PennyLane observable to check sparsity.
+
+        Returns:
+            True if the measurement process will use the sparse data representation.
+        """
+        return isinstance(obs, SparseHamiltonian) or (obs.has_sparse_matrix and not obs.has_matrix)
```

Comment:
Right now we are including both the previous `SparseHamiltonian` and the current `QubitUnitary` from PL here in the same interface right? I haven't started going through the pipeline of pennylane_lightning/core but gut feeling is that there's chance that the operator satisfying this condition can also be caught as a measurement here? 
I understand that eventually they were just the same matrices either Hermitian or Unitary, but I just want to learn a little bit about how the junction between lightning and pennylane works here.

Reply:
This was a mistake. I forgot the second part here after the tests. I was testing for future expansion. For now we are restricting to SparseHamiltonian. Thank you for catching that.